components/sync15/src/engine/sync_engine.rs

```diff
@@ -31,24 +31,28 @@ pub enum EngineSyncAssociation {
 /// The concrete `SyncEngine` implementations
 #[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub enum SyncEngineId {
+    // Note that we've derived PartialOrd etc, which uses lexicographic ordering
+    // of the variants. We leverage that such that the higher priority engines
+    // are listed first.
+    // This order matches desktop.
     Passwords,
-    History,
-    Bookmarks,
     Tabs,
+    Bookmarks,
     Addresses,
     CreditCards,
+    History,
 }
 
 impl SyncEngineId {
     // Iterate over all possible engines
     pub fn iter() -> impl Iterator<Item = SyncEngineId> {
         [
             Self::Passwords,
-            Self::History,
-            Self::Bookmarks,
             Self::Tabs,
+            Self::Bookmarks,
             Self::Addresses,
             Self::CreditCards,
+            Self::History,
```

Comment:
Maybe add a note here that this should be kept in-sync with the enum ordering.

Reply:
I just did that for consistency, but I don't know if it matters - should we guarantee this order is sorted, or should anyone who relies on that actually sort?

Reply:
I maybe slightly prefer guaranteeing the order is sorted, but I can't see a strong reason one way or the other.

Reply:
yeah, fair enough - I added a comment that it's a policy decision and a test to ensure it remains true.